spec/features/state_file/nj/complete_intake_spec.rb

```diff
@@ -0,0 +1,392 @@
+require 'axe-rspec'
+
+RSpec.feature "Completing a state file intake", active_job: true do
+  include MockTwilio
+  include StateFileIntakeHelper
+
+  before do
+    allow_any_instance_of(Routes::StateFileDomain).to receive(:matches?).and_return(true)
+  end
+
+  context "NJ", :flow_explorer_screenshot, js: true do
+
+    def advance_to_start_of_intake(df_persona_name)
+      visit "/"
+      click_on "Start Test NJ"
+
+      expect(page).to have_text I18n.t("state_file.landing_page.edit.nj.title")
+      click_on "Get Started", id: "firstCta"
+
+      continue
+
+      step_through_initial_authentication(contact_preference: :email)
+
+      check "Email"
+      check "Text message"
+      fill_in "Your phone number", with: "+12025551212"
+      click_on "Continue"
+
+      expect(page).to have_text I18n.t('state_file.questions.sms_terms.edit.title')
+      click_on I18n.t("general.accept")
+
+      expect(page).to have_text I18n.t('state_file.questions.terms_and_conditions.edit.title')
+      click_on I18n.t("state_file.questions.terms_and_conditions.edit.accept")
+
+      step_through_df_data_transfer("Transfer #{df_persona_name}")
+
+      expect(page).to have_text I18n.t("state_file.questions.income_review.edit.title")
+      continue
+
+      expect(page).to have_text I18n.t("state_file.questions.nj_eligibility_health_insurance.edit.title")
+      choose I18n.t("general.affirmative")
+      continue
+    end
+
+    def advance_county_and_municipality(county = "Atlantic", municipality = "Atlantic City")
+      select county
+      continue
+
+      select municipality
+      continue
+    end
+
+    def advance_disabled_exemption(selection = false)
+      # disabled exemption
+      page.all(:css, '.white-group').each do |group|
+        within group do
+          choose selection ? I18n.t('general.affirmative') : I18n.t('general.negative')
+        end
+      end
+      continue
+    end
+
+    def advance_college_dependents
+      # college dependents exemption
+      continue
+    end
+
+    def advance_veterans_exemption(selection = false)
+      # veterans exemption
+      page.all(:css, '.white-group').each do |group|
+        within group do
+          choose selection ? I18n.t('general.affirmative') : I18n.t('general.negative')
+        end
+      end
+      continue
+    end
+
+    def advance_medical_expenses(amount: 1000)
+      fill_in I18n.t('state_file.questions.nj_medical_expenses.edit.label', filing_year: filing_year), with: amount
+      continue
+    end
+
+    def continue
+      click_on I18n.t("general.continue")
+    end
+
+    def choose_household_rent_own(household_rent_own)
+      case household_rent_own
+      when "homeowner", "tenant", "both", "neither"
+        choose I18n.t("state_file.questions.nj_household_rent_own.edit.#{household_rent_own}")
+      else
+        throw "not a valid choice"
+      end
+      continue
+    end
+
+    def select_homeowner_eligibility(checkboxes, continue_to_next: true)
```

Comment:
Updated this helper function and the one below by adding `continue_to_next`

Reply:
This pattern will be useful for a11y testing too!